src/Characters.js

```diff
@@ -2,177 +2,169 @@ import React, { Component } from 'react';
 import './styles/main.scss';
 import CharacterInfoCard from './CharacterInfoCard.js';
 import Search from './Search.js';
-import Filter from './Filter.js'
+import Filter from './Filter.js';
 import Universe from './Universe';
-
-// function imagesLoaded() {
-//   const characterGrid = document.querySelector('.characters-grid')
-//   const imgElements = characterGrid.querySelectorAll('div');
-//   imgElements.map( img => {
-//     if (!img.complete) {
-//       return false;
-    
-//   return true;
-//   })
-// }
+import Trie from '@jake.lauer27/autocomplete';
 
 
 class Characters extends Component {
- constructor() {
-   super();
-   this.state = {
-    characterList: [],
-    card: '',
-    characters: [],
-    currentCharacter: '',
-    loading: true,
-    universe: 'all'
-   }
- }
-
- componentDidMount() {
-   fetch('http://whateverly-datasets.herokuapp.com/api/v1/characters')
-     .then(response => response.json())
-     .then(data => {
+  constructor() {
+    super();
+    this.state = {
+      characterList: [],
+      card: '',
+      characters: [],
+      currentCharacter: '',
+      universe: 'all',
+      suggestions: []
+    };
+    this.trie = new Trie();
```

Comment:
Pretty straight on adding to the constructor and populating in componentDidMount.